contracts/truefi2/TrueFiPool2.sol

```diff
@@ -631,7 +631,10 @@ contract TrueFiPool2 is ITrueFiPool2, IPauseableContract, ERC20, UpgradeableClai
      * @dev CreditAgency transfers DebtToken to the pool
      */
     function addDebt(IDebtToken debtToken, uint256 amount) external override {
-        require(msg.sender == address(creditAgency), "TruePool: Only TrueCreditAgency can add debtTokens");
+        require(
+            msg.sender == address(creditAgency) || ftlAgency.loanFactory().isLoanToken(msg.sender),
```

Comment:
I'm really not a fan of using chain calls like this. This definitely breaks some clean code principle. But the alternative is to add loanFactory to the pools (so adding setters and calling them on existing pools). Do you think we can proceed like this or do refactor?

Reply:
Let's do the refactor:
1. add loanFactory to the initializer
2. add setters and call them on existing pools

It's less painful longer term, and we have to upgrade the ImplementationReference anyway to launch this feature.

Reply:
Let's do this in a following PR as it might be another big change